heareval/tasks/speech_commands.py

```diff
@@ -21,111 +23,255 @@
     ResampleSubCorpus,
     SplitTrainTestCorpus,
     SplitTrainTestMetadata,
-    SubsampleCorpus,
+    SubsamplePartition,
     WorkTask,
     ensure_dir,
     filename_to_int_hash,
-    slugify_file_name,
-    which_set,
 )
 
 # Set the task name for all WorkTasks
 WorkTask.task_name = config.TASKNAME
 
+WORDS = ["down", "go", "left", "no", "off", "on", "right", "stop", "up", "yes"]
+BACKGROUND_NOISE = "_background_noise_"
+UNKNOWN = "_unknown_"
+SILENCE = "_silence_"
+
+
+class ExtractArchiveTrain(ExtractArchive):
+    def requires(self):
+        return {
+            "download": DownloadCorpus(
+                url=config.DOWNLOAD_URL, outfile="train-corpus.tar.gz"
+            )
+        }
+
 
-class ExtractArchive(ExtractArchive):
+class ExtractArchiveTest(ExtractArchive):
     def requires(self):
         return {
-            "download": DownloadCorpus(url=config.DOWNLOAD_URL, outfile="corpus.tar.gz")
+            "download": DownloadCorpus(
+                url=config.TEST_DOWNLOAD_URL, outfile="test-corpus.tar.gz"
+            )
         }
 
 
+class GenerateTrainDataset(WorkTask):
+    """
+    Silence / background samples in the train / validation sets need to be
+    created by slicing up longer background samples into 1sec slices
+    """
```

Comment:
Wait what? Why? Is there a cite for this?

If not, can we avoid this or skip this?

Reply:
Ya this is the intention for the dataset. It is shipped with long 1min+ background noise audio for the training and val sets that are supposed to be cut up in 1s chunks. This is the exact method that is used in the tf dataset: https://github.com/tensorflow/datasets/blob/79d56e662a15cd11e1fb3b679e0f978c8041566f/tensorflow_datasets/audio/speech_commands.py#L142

The test set ships with a set of predefined 1s background samples.

Reply:
@jorshi okay add this information to the comment?

Reply:
Will add this as a comment